File: agent-connect-android/app/src/main/java/sg/searchhouse/agentconnect/model/api/userprofile/UserPO.kt
package sg.searchhouse.agentconnect.model.api.userprofile

import com.google.gson.annotations.SerializedName
import sg.searchhouse.agentconnect.model.db.UserEntity
import java.io.Serializable

data class UserPO(
    @SerializedName("id")
    val id: Int,
    @SerializedName("email")
    val email: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("mobileCountryCode")
    val mobileCountryCode: Int,
    @SerializedName("mobileLocalNum")
    val mobileLocalNum: Int,
    @SerializedName("photo")
    val photo: String,
    @SerializedName("companyName")
    val companyName: String,
    @SerializedName("debugAllowed")
    val debugAllowed: Boolean
) : Serializable {

    fun fromUserPOToEntity(): UserEntity {
        return UserEntity(
            id = id,
            email = email,
            name = name,
            mobileCountryCode = mobileCountryCode,
            mobileLocalNum = mobileLocalNum,
            photo = photo,
            companyName = companyName
        )
    }
}

File: srxproperty/src/dataObject/UserPO.js
import {ObjectUtil} from '../utils';

class UserPO {
  constructor(data) {
    if (!ObjectUtil.isEmpty(data)) {
      this.email = data.email;
      this.id = data.id;
      if (data.encryptedId) this.encryptedUserId = data.encryptedId;
      else if (data.encryptedUserId)
        this.encryptedUserId = data.encryptedUserId;
      this.mobileCountryCode = data.mobileCountryCode;
      this.mobileLocalNum = data.mobileLocalNum;
      this.mobileVerified = data.mobileVerified;
      this.name = data.name;
      this.photo = data.photo;
      this.communityAlias = data.communityAlias;
      this.agent = data.agent; //boolean
    }
  }

  getCommunityPostUserName() {
    if (!ObjectUtil.isEmpty(this.communityAlias)) {
      return this.communityAlias;
    } else {
      return this.name;
    }
  }
}

export {UserPO};
